src/components/Tabs.jsx

```diff
@@ -7,19 +7,40 @@ import classNames from 'classnames'
 class Tabs extends React.Component {
     constructor(props, context) {
         super(props, context)
-        this.state = { activeTab: props.defaultTab || 0 }
+        const children = React.Children.toArray(this.props.children)
+
+        this.valueToTab = children.reduce((map, c, i) => {
```

Comment:
This won't work as soon as you add / remove children dynamically, right?

What do you think about adding a function `_getValueForTab(tabs /* children */, value)` which uses `children.find( ... )` to return the correct tab?